src/plugins/imagery/components/ImageryViewLayout.vue

```diff
@@ -79,20 +87,25 @@ export default {
             isPaused: false,
             metadata: {},
             requestCount: 0,
-            timeFormat: ''
+            timeFormat: '',
+            layers: [],
+            visibleLayers: []
         }
     },
     mounted() {
         // set
         this.keystring = this.openmct.objects.makeKeyString(this.domainObject.identifier);
         this.metadata = this.openmct.telemetry.getMetadata(this.domainObject);
-        this.imageFormat = this.openmct.telemetry.getValueFormatter(this.metadata.valuesForHints(['image'])[0]);
+        const metaDataValues = this.metadata.valuesForHints(['image'])[0];
+        this.imageFormat = this.openmct.telemetry.getValueFormatter(metaDataValues);
+        this.loadVisibleLayers(metaDataValues);
         // initialize
         this.timeKey = this.openmct.time.timeSystem().key;
         this.timeFormat = this.openmct.telemetry.getValueFormatter(this.metadata.value(this.timeKey));
         // listen
         this.openmct.time.on('bounds', this.boundsChange);
         this.openmct.time.on('timeSystem', this.timeSystemChange);
+        window.addEventListener('beforeunload', this.persistVisibleLayers);
```

Comment:
This is a creative approach, I like it. The pattern we follow elsewhere though is to just persist immediately on change, so let's keep it consistent with that. That way you catch browser crashes as well.

Reply:
Will do!